kapitan/refs/functions.py

```diff
@@ -20,20 +20,30 @@
 
 
 def eval_func(func_name, ctx, *func_params):
+    """calls specific function which generates the secret"""
+    func_lookup = get_func_lookup()
 
-    func_lookup = {
+    return func_lookup[func_name](ctx, *func_params)
+
+
+def get_func_lookup():
+    """returns the lookup-table for the generator functions"""
+    return {
         "randomstr": randomstr,
         "sha256": sha256,
         "ed25519": ed25519_private_key,
         "rsa": rsa_private_key,
         "rsapublic": rsa_public_key,
         "publickey": public_key,
         "reveal": reveal,
-        "loweralphanum": loweralphanum,
+        "randomint": random_int,
+        "loweralpha": lower_alpha,
+        "loweralphanum": lower_alpha_num,
+        "upperalpha": upper_alpha,
+        "upperalphanum": upper_alpha_num,
+        "alphanumspec": alphanumspec,
     }
 
-    return func_lookup[func_name](ctx, *func_params)
-
 
 def randomstr(ctx, nbytes=""):
```

Comment:
I could update `randomstr` to a more generic solution as well.
It would look like:
```python
def randomstr(ctx, nchars="43"):
    pool = string.ascii_letters + string.digits + "-_"
    generic_alphanum(ctx, nchars, pool)
```
Also we could use `alphanumspec` for that:
```python
def randomstr(ctx, nchars="43"):
    alphanumspec(ctx, nchars, "-_")
```

Reply:
Fixed with latest iteration